File: Baekjoon Online Judge/1080.py
N, M = map(int, input().split())

A = [list(map(int, list(input()))) for _ in range(N)]
B = [list(map(int, list(input()))) for _ in range(N)]


def flip(x, y):
    for i in range(x, x + 3):
        for j in range(y, y + 3):
            A[i][j] = 1 - A[i][j]


def is_equal():
    for i in range(N):
        for j in range(M):
            if A[i][j] != B[i][j]:
                return False
    return True


count = 0

for n in range(N - 2):
    for m in range(M - 2):
        if A[n][m] != B[n][m]:
            flip(n, m)
            count += 1

if is_equal():
    print(count)
else:
    print(-1)

File: Baekjoon Online Judge/1080.cpp
#include <stdio.h>

int N, M;
int A[50][50];
int B[50][50];

int dy[4] = { -1, 0, 1, 0 };
int dx[4] = { 0, -1, 0, 1 };

void flip(int y, int x) {
	for (int i = 0; i < 3; ++i)
		for (int j = 0; j < 3; ++j)
			A[y + i][x + j] = 1 - A[y + i][x + j];
}

bool equals() {
	for (int i = 0; i < N; ++i)
		for (int j = 0; j < M; ++j)
			if (A[i][j] != B[i][j])
				return false;

	return true;
}

int main() {
	int i, j, count(0);

	scanf("%d %d", &N, &M);

	for (i = 0; i < N; ++i)
		for (j = 0; j < M; ++j)
			scanf("%1d", &A[i][j]);

	for (i = 0; i < N; ++i)
		for (j = 0; j < M; ++j)
			scanf("%1d", &B[i][j]);

	for (i = 0; i < N - 2; ++i)
		for (j = 0; j < M - 2; ++j)
			if (A[i][j] != B[i][j]) {
				flip(i, j);
				++count;
			}

	if (equals())
		printf("%d", count);
	else
		printf("-1");

	return 0;
}
